### PR TITLE
[risk=no][RW-5132] Disable multiple clicks of Confirm for Create/Edit Workspace

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test:local`
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 1]
```diff
@@ -934,13 +934,29 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
     render() {
       const {enableBillingUpgrade} = serverConfigStore.getValue();
       const {
-        populationChecked,
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:937]: none of these changes are necessary to the PR.  I can revert if you don't like them.

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 2]
```diff
@@ -1374,12 +1389,18 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
               <div>You can also make changes to your answers after you create your workspace.</div>
             </ModalBody>
             <ModalFooter>
-              <Button type='secondary' style={{marginRight: '1rem'}}
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1377]: Here begins the actual fix.  "disabled if errors or loading"

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx [hunk 3]
```diff
@@ -216,6 +216,29 @@ describe('WorkspaceEdit', () => {
     expect(navigate).toHaveBeenCalledTimes(1);
   });
 
+  // regression test for RW-5132
+  it('prevents multiple Workspace creations via the same confirmation dialog', async() => {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:220]: confirmed that this test fails with the previous version of the code

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 2 above)
Comment (jaycarlton) [reply ui/src/app/pages/workspace/workspace-edit.tsx:1377]: Nice. It would be handy to have a reusable pattern for this kind of control.

**Pull request merged**